examples/dataproc-idle-shutdown/isIdleJob.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+# Copyright 2019 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+readonly MY_CLUSTER_NAME="$(/usr/share/google/get_metadata_value attributes/dataproc-cluster-name)"
+readonly MAX_IDLE_SECONDS_KEY="${MY_CLUSTER_NAME}_maxIdleSeconds"
+readonly DATAPROC_PERSIST_DIAG_TARBALL_KEY="${MY_CLUSTER_NAME}_persistDiagnosticTarball"
+readonly KEY_PROCESS_LIST_KEY="${MY_CLUSTER_NAME}_keyProcessList"
+readonly ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
+readonly PERSIST_DIAG_TARBALL_TRUE="TRUE"
+readonly PERSIST_DIAG_TARBALL_FALSE="FALSE"
+
+function checkMaster() {
```

Comment:
Updated

Reply:
The above will only be TRUE if the cluster is high availability. In that case, it will be true for all machines. I actually don't think it makes sense to use an idle process shutdown for a high availability cluster in any case. I'm going to update the code to ONLY run scripts in a non HA environment.